src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NFloat.cs

```diff
@@ -701,5 +714,664 @@ public int CompareTo(object? obj)
+
+        // /// <inheritdoc cref="IExponentialFunctions{TSelf}.Exp10(TSelf)" />
+        // public static NFloat Exp10(NFloat x) => new NFloat(NativeType.Exp10(x._value))(x);
+
+        // /// <inheritdoc cref="IExponentialFunctions{TSelf}.Exp10M1(TSelf)" />
+        // public static NFloat Exp10M1(NFloat x) => new NFloat(NativeType.Exp10M1(x._value));
+
+        //
+        // IFloatingPoint
+        //
+
+        /// <inheritdoc cref="IFloatingPoint{TSelf}.Ceiling(TSelf)" />
+        public static NFloat Ceiling(NFloat x) => new NFloat(NativeType.Ceiling(x._value));
+
+        /// <inheritdoc cref="IFloatingPoint{TSelf}.Floor(TSelf)" />
+        public static NFloat Floor(NFloat x) => new NFloat(NativeType.Floor(x._value));
+
+        /// <inheritdoc cref="IFloatingPoint{TSelf}.Round(TSelf)" />
+        public static NFloat Round(NFloat x) => new NFloat(NativeType.Round(x._value));
+
+        /// <inheritdoc cref="IFloatingPoint{TSelf}.Round(TSelf, int)" />
+        public static NFloat Round(NFloat x, int digits) => new NFloat(NativeType.Round(x._value, digits));
+
+        /// <inheritdoc cref="IFloatingPoint{TSelf}.Round(TSelf, MidpointRounding)" />
+        public static NFloat Round(NFloat x, MidpointRounding mode) => new NFloat(NativeType.Round(x._value, mode));
+
+        /// <inheritdoc cref="IFloatingPoint{TSelf}.Round(TSelf, int, MidpointRounding)" />
+        public static NFloat Round(NFloat x, int digits, MidpointRounding mode) => new NFloat(NativeType.Round(x._value, digits, mode));
+
+        /// <inheritdoc cref="IFloatingPoint{TSelf}.Truncate(TSelf)" />
+        public static NFloat Truncate(NFloat x) => new NFloat(NativeType.Truncate(x._value));
+
+        /// <inheritdoc cref="IFloatingPoint{TSelf}.GetExponentShortestBitLength()" />
+        long IFloatingPoint<NFloat>.GetExponentShortestBitLength()
+        {
+            NativeExponentType exponent = _value.Exponent;
+
+            if (exponent >= 0)
+            {
+                return (sizeof(NativeExponentType) * 8) - NativeExponentType.LeadingZeroCount(exponent);
+            }
+            else
+            {
+                return (sizeof(NativeExponentType) * 8) + 1 - NativeExponentType.LeadingZeroCount((NativeExponentType)(~exponent));
+            }
+        }
+
+        /// <inheritdoc cref="IFloatingPoint{TSelf}.GetExponentByteCount()" />
+        int IFloatingPoint<NFloat>.GetExponentByteCount() => sizeof(NativeExponentType);
+
+        /// <inheritdoc cref="IFloatingPoint{TSelf}.TryWriteExponentLittleEndian(Span{byte}, out int)" />
+        bool IFloatingPoint<NFloat>.TryWriteExponentLittleEndian(Span<byte> destination, out int bytesWritten)
+        {
+            if (destination.Length >= sizeof(NativeExponentType))
+            {
+                NativeExponentType exponent = _value.Exponent;
+                Unsafe.WriteUnaligned(ref MemoryMarshal.GetReference(destination), exponent);
```

Comment:
This write isn't endian-aware when it probably should be.

Reply:
Will get up a PR handling this.